cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/service/UserService.kt

```diff
@@ -0,0 +1,47 @@
+package com.cognifide.cogboard.config.service
+
+import com.cognifide.cogboard.CogboardConstants.Companion.PROP_CURRENT_PASSWORD
+import com.cognifide.cogboard.CogboardConstants.Companion.PROP_NEW_PASSWORD
+import com.cognifide.cogboard.CogboardConstants.Companion.PROP_PASSWORD
+import com.cognifide.cogboard.CogboardConstants.Companion.PROP_USER
+import com.cognifide.cogboard.config.model.Admin
+import com.cognifide.cogboard.storage.Storage
+import com.cognifide.cogboard.storage.VolumeStorageFactory.admin
+import io.vertx.core.json.JsonObject
+
+class UserService(
+    private val storage: Storage = admin(),
+    private val config: JsonObject = storage.loadConfig()
+) {
+    fun loadConfig(): JsonObject = admin().loadConfig()
+
+    fun save(newLoginData: JsonObject): JsonObject {
+        val wrongPassMsg = config.getString("wrongPassMsg") ?: "Please, enter correct Password"
+        val admin = newLoginData.toAdmin()
+        val currentPassword = newLoginData.getString(PROP_CURRENT_PASSWORD)
+        return if (checkCurrentPassword(currentPassword)) {
+            update(admin)
+            response()
+        } else response(wrongPassMsg)
+    }
+
+    private fun checkCurrentPassword(currentPassword: String): Boolean {
```

Comment:
`checkCurrentPassword` is not the best description of what the method is actually doing.
`confirmPassword` maybe or smth like this

Reply:
done